src/modules/commander/state_machine_helper.cpp

```diff
@@ -766,6 +785,14 @@ bool check_invalid_pos_nav_state(vehicle_status_s &status, bool old_failsafe, or
 	}
 
 	if (fallback_required) {
+
```

Comment:
For the interested reviewer: This code had to be moved up here because it computes **status.failsafe_timestamp** which is then used in the logic below.